sql/logic_test.go

```diff
@@ -207,9 +208,36 @@ func (t *logicTest) setUser(user string) {
 		t.Fatal(err)
 	}
 	certDir := filepath.Join(filepath.Dir(dir), "resource", security.EmbeddedCertsDir)
+
+	certPath := security.ClientCertPath(certDir, user)
+	keyPath := security.ClientKeyPath(certDir, user)
+
+	// `github.com/lib/pq` requires that private key file permissions are
+	// "u=rw (0600) or less".
+	f, err := ioutil.TempFile(os.TempDir(), "roach_logic_test_key")
+	if err != nil {
+		t.Fatal(err)
+	}
+	tmpKeyPath := f.Name()
+	defer os.Remove(tmpKeyPath)
+
+	if err := f.Close(); err != nil {
+		t.Fatal(err)
+	}
+	key, err := ioutil.ReadFile(keyPath)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := ioutil.WriteFile(tmpKeyPath, key, 0600); err != nil {
+		t.Fatal(err)
+	}
+
 	db, err := sql.Open("postgres",
-		fmt.Sprintf("sslmode=verify-full sslrootcert=%s user=%s host=%s port=%s",
-			filepath.Join(certDir, "ca.crt"), user, host, port))
+		fmt.Sprintf("sslmode=verify-full sslrootcert='%s' sslcert='%s' sslkey='%s' user=%s host=%s port=%s",
```

Comment:
Adding quotes works for spaces, but filenames can also contain quote characters (admittedly less likely). Assuming the syntax used here allows some sort of escape syntax we should use it.